tasks/lammps/data.py

```diff
@@ -1,31 +1,50 @@
 from invoke import task
-from os.path import exists
+from os.path import exists, join
 import requests
 
-from tasks.lammps.env import LAMMPS_DATA_FILE
+from tasks.lammps.env import (
+    BENCHMARKS,
+    LAMMPS_DIR,
+    LAMMPS_FAASM_DATA_PREFIX,
+)
 from tasks.util.faasm import get_faasm_upload_host_port
 
-FAASM_LOCAL_SHARED_DIR = "/usr/local/faasm/shared"
-
-RELATIVE_PATH = "lammps-data/in.controller"
-
 
 @task(default=True)
-def upload(ctx):
+def upload(ctx, bench):
     """
-    Upload LAMMPS data to Faasm
+    Upload LAMMPS benchmark data to Faasm
     """
-    if not exists(LAMMPS_DATA_FILE):
-        print("Did not find data at {}".format(LAMMPS_DATA_FILE))
-        exit(1)
+    if bench not in BENCHMARKS:
+        print("Unrecognized benchmark: {}".format(bench))
+        print(
+            "The supported LAMMPS benchmarks are: {}".format(BENCHMARKS.keys())
+        )
+
+    _bench = BENCHMARKS[bench]
 
     host, port = get_faasm_upload_host_port()
     url = "http://{}:{}/file".format(host, port)
-    print("Uploading LAMMPS data to {}".format(url))
-    response = requests.put(
-        url,
-        data=open(LAMMPS_DATA_FILE, "rb"),
-        headers={"FilePath": RELATIVE_PATH},
-    )
-
-    print("Response {}: {}".format(response.status_code, response.text))
+
+    # Upload all data corresponding to the benchmark
+    for data in _bench["data"]:
```

Comment:
It looks like some of the benchmarks don't have a `"data"` entry, you could either add an empty one for each, or do something like:

```python
for data in _bench.get("data", list()):
```